packages/db-base/lib/inMemFileDB.js

```diff
@@ -264,50 +271,81 @@ class InMemoryFileDB {
         return text;
     }
 
+    /**
+     * Handle saving the dataset incl. backups
+     */
     saveState() {
+        const jsonString = this.saveDataset();
+
+        if (!this.settings.backup.disabled && jsonString) {
+            this.saveBackup(jsonString);
+        }
+
+        if (this.stateTimer) {
+            clearTimeout(this.stateTimer);
+            this.stateTimer = null;
+        }
+    }
+
+    /**
+     * Saves the dataset into File incl. handling of a fallback backup file
+     *
+     * @returns {string} JSON string of the complete dataset to also be stored into a compressed backup file
+     */
+    saveDataset() {
```

Comment:
Also - async save methods maybe?

Reply:
Only called by Timeouts or on destroy if dB. I do not really see a Benefit if async. Or?!

Reply:
Ok, I change to async

Reply:
done. the compressed backup is done via a stream and so asny itself ... so I would leave that